File: app/src/main/java/com/aslansari/hypocoin/repository/restapi/Asset.java
package com.aslansari.hypocoin.repository.restapi;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Asset<T> {

    @SerializedName(value = "status")
    public Status status;
    @SerializedName(value = "data")
    public List<T> assets;

}

File: app/src/main/java/com/aslansari/hypocoin/repository/restapi/Asset.kt
package com.aslansari.hypocoin.repository.restapi

import com.google.gson.annotations.SerializedName

data class Asset<T>(
    @SerializedName(value = "status")
    var status: Status? = null,
    @SerializedName(value = "data")
    var assets: List<T>? = null
)
